server.py

```diff
@@ -1,36 +1,376 @@
+import argparse
+import json
+import logging
+import os
+import sys
+from collections import namedtuple
+from typing import Collection, Dict, List, Optional, Tuple, Union
+
 import sanic
-import functools
+import sanic.request
+import sanic.response
+
+logger = logging.getLogger(__name__)
 
 app = sanic.Sanic("hello_example")
 
-app.static('/', 'build/')
+app.static("/", "build/")
+
+
+# JSON type for coach data.
+CoachJSON = Dict[str, Union[str, int, bool, float, "CoachJSON"]]
+
+# ------------------------------------------------------------------------------
+# Database
+# ------------------------------------------------------------------------------
+
+
+class Coach(
+    namedtuple(
+        "_Coach",
+        [
+            "id",
+            "name",
+            "bio",
+            "available",
+            "birth_year",
+            "gender",
+            "languages",
+            "need",
+            "rights",
+            "housing",
+        ],
+    )
+):
+    """A coach object - an entry in the database."""
+
+    def __new__(
+        cls,
+        *,
+        id: Optional[int] = None,
+        name: str,
+        bio: str,
+        available: bool = True,
+        birth_year: int,
+        gender: str,
+        languages: Dict[str, int],
+        need: Collection[int],
+        rights: Collection[int],
+        housing: Collection[int],
+    ):
+        # TODO: Validate args.
```

Comment:
@@@ TODO would be easier to search for if that's okay!

Reply:
*I'm happy to do this, but first I'm treating you to a bit of an essay on my thoughts here ;) just give me a "understood, but still prefer @@@" if you want!*

Eh my impression is that @@@ is a bit of an Ensoftism, TODO seems to be the most common in the industry (or perhaps 'FIXME', although that has a slightly different feel). Personally I've started using @@@ and TODO in my own code with the following meaning:
- @@@: Should never be merged into a branch that claims to be stable - this is a placeholder for something that's broken and needs fixing.
- TODO: Something that could be improved but doesn't need doing immediately. I only use this in projects that are incremental (i.e. all personal projects, no official Ensoft work). I.e. it's a way to avoid getting stuck in a rabbit hole, making a note rather than working out details there and then. This allows time to mull it over. If it ever ends up being an issue the 'TODO' serves as an indicator for what might need doing. This practice is personal taste (some would argue you should instead have a skipped failing unit test).

Specifically in this case, I'm not sure this deserves an @@@, and if you don't like 'TODO's then I'd be tempted to remove this comment entirely (perhaps move to a GH issue, although it feels a bit too low-level for a GH issue on this early-stage project).

Reply:
Nice - I like that system, let's go for that. I'll start using it in my branch too!